src/screens/AccountDetails.js

```diff
@@ -73,14 +73,48 @@ class AccountDetailsView extends React.PureComponent {
 		);
 	};
 
-	onOptionSelect = value => {
+	onOptionSelect = async value => {
 		const navigate = this.props.navigation.navigate;
+		const accounts = this.props.accounts;
 
 		if (value !== 'AccountEdit') {
-			navigate('AccountUnlock', {
-				next: value,
-				onDelete: this.onDelete
-			});
+			if (accounts.getSelected().biometricEnabled) {
+				try {
+					await accounts.unlockAccountWithBiometric(accounts.getSelectedKey());
+					if (value === 'AccountDelete') {
+						this.onDelete();
+					} else if (value === 'AccountBiometric') {
+						await accounts.disableBiometric(accounts.getSelectedKey());
+					} else {
+						navigate(value);
+					}
+				} catch (e) {
+					console.log(e);
+					Alert.alert('Biometric Error', e.message, [
+						{
+							onDismiss: () => {
+								navigate('AccountUnlock', {
```

Comment:
better wrap following  block into function
```
								navigate('AccountUnlock', {
									next: value,
									onDelete: this.onDelete
								});
```

and put the alert function into `alertUtils.js`